### PR TITLE
add checks to forms and general data

### PR DESCRIPTION
**Pull request by karladler**
- disable sharing when not owner of container
- add access check to forms and disable edit mode of not allowed to write
- set dbcp form icon to public
- [CORE-965]

**do not merge before CORE 964 is merged**

[CORE-965]: https://evannetwork.atlassian.net/browse/CORE-965

### CODE REVIEW

File: dapps/digital-twin-detail/src/components/container/Entry.ts [hunk 1]
```diff
@@ -37,6 +37,29 @@ export default class ContainerEntryComponent extends mixins(EvanComponent) {
...
+      return 'readWrite';
+    }
+
+    return 'none';
+  }
+
+  /**
+   * Returns true if the current account is the owner of the container.
+   */
+  get isOwner(): boolean {
```
Comment (Tschuck) [inline dapps/digital-twin-detail/src/components/container/Entry.ts]: Also the `isOwner` flag could be stored in the `DAppContainer`.

**Comment by datrinh** [comment]
> We could add `readPermissions` / `writePermissions` / `isOwner` flags for the current user to the `DAppContainer`.
> 
> As a result of this, we could just use the maps for enabling / disabling the views / interactions without a `access` / `isOwner` getter. We also can remove duplicated implementations in multiple components.
> 
> * `container.permissions.read[this.name]`
> * `container.permissions.write[this.name]`
> * `container.permissions.isOwner`

Similar to Tobi's comment, I think this is the perfect place to apply smart/dumb component principle. I tried to keep the ListComponent as dumb as possible. Just pass it the properties it needs and do the logic from a smart component (container in this case). ListComp becomes very easy to test in this way.

File: dapps/digital-twin-detail/src/components/container/List.ts [hunk 2]
```diff
@@ -104,7 +119,7 @@ export default class ContainerListComponent extends mixins(EvanComponent) {
       : `${fileList.files.length} ${this.$t('_twin-detail.data.files')}`;
   }
 
-  getValues(): any[] {
+  get values(): any[] {
```
Comment (Tschuck) [inline dapps/digital-twin-detail/src/components/container/List.ts]: Already tried this during implementation, but got problems with the refreshing on dispatcher finish. Can you check, if the list gets correctly refreshed, when the dispatcher has finished?

File: core/digital-twin-lib/src/DAppContainer.ts [hunk 3]
```diff
@@ -198,17 +220,20 @@ class DAppContainer extends Container {
    * @param      {string}  entriesToLoad  load only specific entriess
    */
   public async loadEntryValues(entriesToLoad?: string[]): Promise<void> {
+    // TODO: why is key for type returned here? For what is it used?
```
Comment (Tschuck) [inline core/digital-twin-lib/src/DAppContainer.ts]: Type is a general property on every DataContainer. For our case, we can just remove it.

File: core/digital-twin-lib/src/DAppContainer.ts [hunk 4]
```diff
@@ -298,12 +311,34 @@ class DAppContainer extends Container {
...
-      read: [],
-      readWrite: [],
-      ...await this.getContainerShareConfigForAccount(this.runtime.activeAccount),
+  /**
+   * Returns permissions object with the same structure as the plugin.
+   * Adds `type {read: true}` to permissions, because it's omitted in getContainerShareConfigForAccount() function.
+   */
+  async getPermissions(): Promise<Permissions> {
+    const {
+      read = [],
```
Comment (datrinh) [inline core/digital-twin-lib/src/DAppContainer.ts]: What does this syntax do? Init a const var with `[]` and then overwrite it? Does that even work?

File: core/digital-twin-lib/src/DAppContainer.ts [hunk 5]
```diff
@@ -298,12 +311,34 @@ class DAppContainer extends Container {
...
+
+    const permissions = {
+      type: {
+        read: true,
+        readWrite: false,
+        removeListEntries: false,
+      },
     };
+
+    read.concat(readWrite).forEach((item) => {
```
Comment (datrinh) [inline core/digital-twin-lib/src/DAppContainer.ts]: Does this include duplicates then? Iterate over a Set to remove them?

File: dapps/digital-twin-detail/src/components/container/List.ts [hunk 6]
```diff
@@ -45,6 +45,21 @@ export default class ContainerListComponent extends mixins(EvanComponent) {
 
   selectedValue = null;
 
+  /**
+   * Get the total number of list entries or `?` if we don't have access to the list.
+   */
+  get totalEntries(): number|string {
```
Comment (datrinh) [inline dapps/digital-twin-detail/src/components/container/List.ts]: Do both Tobi and you prefer `number|string` over `number | string`? I prefer the latter, but I would bow to the majority :)

File: dapps/digital-twin-detail/src/components/container/List.vue [hunk 7]
```diff
@@ -59,6 +60,7 @@
             >{{ '_twin-detail.data.list.show-all' | translate }}</span>
           </b-dropdown-item>
           <b-dropdown-item
+            v-if="isEditable"
```
Comment (datrinh) [inline dapps/digital-twin-detail/src/components/container/List.vue]: I think we agreed on greying it out yesterday? We can disable it just like `Show All`.

File: core/digital-twin-lib/src/DAppContainer.ts (hunk 5 above)
Comment (karladler) [reply core/digital-twin-lib/src/DAppContainer.ts]: it shouldn't include any duplicates as long as the bcc is working properly.

File: core/digital-twin-lib/src/DAppContainer.ts (hunk 4 above)
Comment (karladler) [reply core/digital-twin-lib/src/DAppContainer.ts]: It's destructuring assignment with default var, if the returned 'read' is undefined, it's set to empty array.

File: dapps/digital-twin-detail/src/components/container/List.ts (hunk 6 above)
Comment (karladler) [reply dapps/digital-twin-detail/src/components/container/List.ts]: I didn't thought about, personally I like it without the spaces more, but everybody else seems to use spaces. So let's stick to the spaces.

**Pull request merged**